File: javaQuarkusShip/src/main/java/pl/battleships/javaquarkusship/config/GameConfig.java
package pl.battleships.javaquarkusship.config;

import io.quarkus.arc.DefaultBean;
import pl.battleships.core.api.BattleshipGame;
import pl.battleships.core.api.BattleshipGameImpl;
import pl.battleships.core.api.HistoryProvider;
import pl.battleships.core.api.ShotHandler;
import pl.battleships.javaquarkusship.mapper.GameMapper;


import javax.enterprise.context.ApplicationScoped;
import javax.enterprise.inject.Produces;

public class GameConfig {

    @Produces
    @ApplicationScoped
    @DefaultBean
    public GameMapper mapper() {
        return GameMapper.INSTANCE;
    }

    @Produces
    @ApplicationScoped
    public BattleshipGame battleshipGame(HistoryProvider historyProvider, ShotHandler shotHandler) {
        return new BattleshipGameImpl(historyProvider, shotHandler);
    }
}

File: kotlinQuarkusShip/src/main/kotlin/pl/battleships/kotlinquarkusship/config/GameConfig.kt
package pl.battleships.kotlinquarkusship.config

import pl.battleships.core.api.BattleshipGame
import pl.battleships.core.api.BattleshipGameImpl
import pl.battleships.core.api.HistoryProvider
import pl.battleships.core.api.ShotHandler
import javax.enterprise.context.ApplicationScoped
import javax.enterprise.inject.Produces

class GameConfig {

    @Produces
    @ApplicationScoped
    fun battleshipGame(historyProvider: HistoryProvider, shotHandler: ShotHandler): BattleshipGame {
        return BattleshipGameImpl(historyProvider, shotHandler)
    }
}